README.md

```diff
@@ -257,6 +257,105 @@ replacing the type converters with something else. These elements, in possible
 conjunction with elements from the Hierarchy API, are flexible enough to build
 anything you need.
 
+## Special Type Parsing and Casting
+
+We use the `cast` library for casting values read in from configurations into their go types. The `cast` library 
+falls back to JSON for complex types expressed as string values. Here are some examples of how we parse different types:
+
+**[]string:**
+For a given configuration
+```go
+type Config struct {
+    TheSlice *[]string
+}
+```
+The values in the following examples will all be parsed as a string slice.
+```yaml
+theslice:
+  - "a"
+  - "b"
+  - "c"
+```
+
+```shell
+SOME_ENV_VAR="a b c"`
```

Comment:
task: I believe we need some words here to describe what's happening. It isn't clear if we are demonstrating how to set the appropriate environment variable for loading directly or if this is an example of how the environment variable interpolation feature of JSON/YAML can be used. For example,
```shell
CONFIG_THESLICE="a b c"
```
is the appropriate way to configure for using the ENV loader but any variable name could be used in conjunction with the JSON or YAML files.